dashboard/app/controllers/ai_diff_controller.rb

```diff
@@ -31,7 +31,7 @@ def chat_completion
 
     @unit = Unit.find_by(id: @lesson.script_id)
     @unit_group = @unit.unit_groups.first
-    course_name = CourseOffering.find_by(id: @unit_group.course_version.course_offering_id).display_name
+    course_name = CourseOffering.find_by(id: @unit_group&.course_version&.course_offering_id)&.display_name
```

Comment:
What's the difference here? Just so I know what I did incorrectly/ how to tell when I need the `&`

Reply:
The `&.` is analogous to the `?.` in JavaScript. This way if `@unit_group` or any of the other things in the chain are nil the `course_name` will be set to nil instead of raising an exception.

Reply:
Confession: I sing "safe operator" like [this](https://www.youtube.com/watch?v=rS7Va0sBYAM) in my head whenever I use these...